qiskit/ignis/verification/randomized_benchmarking/dihedral.py

```diff
@@ -74,27 +75,22 @@ def mul_monomial(self, indices):
         """
         length = len(indices)
         assert length < 4, "no term!"
-        assert True not in [x < 0 or x >= self.n_vars for x in indices], \
-            "indices out of bounds!"
-        assert False not in [indices[i] < indices[i+1]
-                             for i in range(length-1)], \
-            "indices non-increasing!"
+        assert ((np.array(indices) < 0).all() is not False) and \
+               ((np.array(indices) >= self.n_vars).all() is not False), "indices out of bounds!"
+        if length > 1:
+            assert (np.diff(np.array(indices)) > 0), "indices non-increasing!"
```

Comment:
You forgot `.all()`. Also, again, no need to convert to an `np.array()` every time.